include/envoy/thread_local/thread_local.h

```diff
@@ -6,30 +6,13 @@
 
 #include "envoy/common/pure.h"
 #include "envoy/event/dispatcher.h"
+#include "envoy/thread_local/thread_local_object.h"
+
```

Comment:
I think that the dispatcher could be forward declared and dispatcher interface library depedency removed as a dependency if that would make some of these changes easier.